src/core_init_atmosphere/mpas_init_atm_gwd.F

```diff
@@ -314,14 +334,17 @@ function read_global_30s_topo(path, sub_path) result(iErr)
       integer, parameter :: tile_y = 1200       ! y-dimension of each tile of global 30-arc-second topography
       integer, parameter :: tile_bdr = 3        ! number of layers of border/halo points surrounding each tile
 
-      integer :: istatus
+      integer (c_int) :: istatus
       integer :: ix, iy, ishift, ix_shift
-      integer :: isigned, endian, wordsize, nx, ny, nz
-      real (kind=R4KIND) :: scalefactor
-      real (kind=R4KIND), dimension(:,:,:), allocatable :: tile
+      integer (c_int) :: isigned, endian, wordsize, nx, ny, nz
+      real (c_float) :: scalefactor
+      real (c_float), dimension(:,:,:), pointer, contiguous :: tile
+      type (c_ptr) :: tile_ptr
       character(len=StrKIND) :: filename
+      character(len=StrKIND, kind=c_char) :: c_filename
```

Comment:
I think `c_filename` should be have size `StrKIND+1` to allow for the addition of a C null character in the `mpas_f_to_c_string` routine. Also, I think `c_filename` should be declared as an array, i.e.,
```
character(kind=c_char), dimension(StrKIND+1) :: c_filename
```
I'll check on this last point (declaration as an array), though, as my memory of interoperability of C character arrays may be fuzzy.